shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -501,6 +519,55 @@ void Engine::OnDebugWireframeSettingsChanged(bool enabled) {
       });
 }
 
+void Engine::OnCreateViewMethodCall(int64_t view_id,
+                                    bool hit_testable,
+                                    bool focusable) {
+  if (!shell_) {
+    return;
+  }
+
+  shell_->GetTaskRunners().GetRasterTaskRunner()->PostTask(
+      [rasterizer = shell_->GetRasterizer(), view_id, hit_testable,
+       focusable]() {
+        if (rasterizer) {
+          auto compositor_context =
+              reinterpret_cast<flutter_runner::CompositorContext*>(
+                  rasterizer->compositor_context());
+          compositor_context->OnCreateView(view_id, hit_testable, focusable);
+        }
+      });
+}
+
+void Engine::OnDestroyViewMethodCall(int64_t view_id) {
+  if (!shell_) {
+    return;
+  }
+
+  shell_->GetTaskRunners().GetRasterTaskRunner()->PostTask(
+      [rasterizer = shell_->GetRasterizer(), view_id]() {
+        if (rasterizer) {
+          auto compositor_context =
+              reinterpret_cast<flutter_runner::CompositorContext*>(
+                  rasterizer->compositor_context());
+          compositor_context->OnDestroyView(view_id);
+        }
+      });
+}
+
+flutter::ExternalViewEmbedder* Engine::GetViewEmbedder() {
```

Comment:
can we make this a `shared_ptr`?

Reply:
My next CL is going to move the ExternalViewEmbedder into the engine and make it a shared_ptr there, so it makes sense to just do it now